lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -323,46 +397,86 @@ def create_eni(connection, vpc_id, module):
+    purge_tags = module.params.get("purge_tags")
+
+    if name:
+        tags["Name"] = name
 
     try:
-        eni = connection.create_network_interface(subnet_id, private_ip_address, description, security_groups)
+        args = {"SubnetId": subnet_id}
+        if private_ip_address:
+            args["PrivateIpAddress"] = private_ip_address
+        if description:
+            args["Description"] = description
+        if len(security_groups) > 0:
+            args["Groups"] = security_groups
+        eni_dict = connection.create_network_interface(**args)
+        eni = eni_dict["NetworkInterface"]
         if attached and instance_id is not None:
             try:
-                eni.attach(instance_id, device_index)
-            except BotoServerError:
-                eni.delete()
+                connection.attach_network_interface(
+                    InstanceId=instance_id,
+                    DeviceIndex=device_index,
+                    NetworkInterfaceId=eni["NetworkInterfaceId"]
+                )
+            except botocore.exceptions.ClientError:
+                connection.delete_network_interface(eni["NetworkInterfaceId"])
                 raise
             # Wait to allow creation / attachment to finish
-            wait_for_eni(eni, "attached")
-            eni.update()
+            wait_for_eni(connection, "attached", module, eni)
+            eni = uniquely_find_eni(connection, module, eni)
 
         if secondary_private_ip_address_count is not None:
             try:
-                connection.assign_private_ip_addresses(network_interface_id=eni.id, secondary_private_ip_address_count=secondary_private_ip_address_count)
-            except BotoServerError:
-                eni.delete()
+                connection.assign_private_ip_addresses(
+                    NetworkInterfaceId=eni["NetworkInterfaceId"],
+                    SecondaryPrivateIpAddressCount=secondary_private_ip_address_count
+                )
+                eni = uniquely_find_eni(connection, module, eni)
+            except botocore.exceptions.ClientError:
+                connection.delete_network_interface(eni["NetworkInterfaceId"])
                 raise
 
         if secondary_private_ip_addresses is not None:
             try:
-                connection.assign_private_ip_addresses(network_interface_id=eni.id, private_ip_addresses=secondary_private_ip_addresses)
-            except BotoServerError:
-                eni.delete()
+                connection.assign_private_ip_addresses(
+                    NetworkInterfaceId=eni["NetworkInterfaceId"],
+                    PrivateIpAddresses=secondary_private_ip_addresses
+                )
+                eni = uniquely_find_eni(connection, module, eni)
+            except botocore.exceptions.ClientError:
+                connection.delete_network_interface(eni["NetworkInterfaceId"])
                 raise
 
+        eni = uniquely_find_eni(connection, module, eni)
```

Comment:
If all the other code paths call this, this line is unnecessary.

Reply:
Good catch.  I'll remove it.